source/opt/ir_context.cpp

```diff
@@ -95,6 +95,10 @@ Instruction* IRContext::KillInst(ir::Instruction* inst) {
     }
   }
 
+  if (type_mgr_) {
+    type_mgr_->RemoveId(inst->result_id());
```

Comment:
We will be killing a to of instructions that are not type.  Since the type manager is always valid this will mean a lot of hash table look ups.  Would if be better to have a quick check of the opcode?  If it is not OpType* then don't try to remove it.